### PR TITLE
ONE: Toxic Keyword

### LINKED ISSUE #2116: ONE: Toxic
Toxic N seems to work similar to Infect that it works as static, but also similar to Poisonous N that it uses the Keyword amount for the effect.

use `getKeywordMagnitude` to get the combined values of their keywords.

For the interaction with Infect, is a card has both and if it should cause 1 or 2 AddCounter Trigger on Player we probably need the Mechanics video.

### LINKED ISSUE #2210: Venser, Corpse Puppet
> Venser, Corpse Puppet
> U B
> Legendary Creature - Phyrexian Zombie Wizard
> Lifelink, toxic 1
> Whenever you proliferate, choose one - 
> * If you don't control a creature named The Hollow Sentinel,  create The Hollow Sentinel, a legendary 3/3 colorless Phyrexian Golem  artifact creature token. 
> * Target artifact creature you control gains flying and lifelink until end of turn.

`Whenever you proliferate` Trigger

should be pretty strait forward

### PR DESCRIPTION
**Pull request by Hanmac**
Closes #2116 
Closes #2210

First card with Toxic

@Agetian does it need more AI logic? Like in places with Infect?

### CODE REVIEW

**Comment by Agetian** [comment]
Hmm, it might need to be hooked up to the places that process poison for the AI, for memory, there were some places in AiAttackController/AiBlockController that dealt with Poison, and probably various other stuff. Can anyone take a look please? (Currently out of town for the holiday season...)

**Comment by Agetian** [comment]
Another good place to look at would be ComputerUtilCombat.poisonIfUnblocked (and possibly other similar routines I may not remember atm :) )

**Comment by tool4ever** [comment]
👍

**Review by Agetian — DISMISSED** [review]

**Comment by Agetian** [comment]
Looks good to me :)

**Comment by Hanmac** [comment]
i think the other Toxic cards can be added in later MR

File: forge-gui/res/cardsfolder/upcoming/venser_corpse_puppet.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Name:Venser, Corpse Puppet
+ManaCost:U B
+Types:Legendary Creature Phyrexian Zombie Wizard
+PT:1/3
+K:Lifelink
+K:Toxic:1
+T:Mode$ Proliferate | ValidPlayer$ You | Execute$ TrigChoose | TriggerDescription$ Whenever you proliferate, ABILITY
+SVar:TrigChoose:DB$ Charm | Choices$ DBToken,DBPump
+SVar:DBToken:DB$ Token | TokenAmount$ 1 | TokenScript$ the_hollow_sentinel | TokenOwner$ You | ConditionPresent$ Creature.YouCtrl+namedThe Hollow Sentinel | PresentCompare$ EQ0 | SpellDescription$ If you don't control a creature named The Hollow Sentinel, create The Hollow Sentinel, a legendary 3/3 colorless Phyrexian Golem artifact creature token.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/venser_corpse_puppet.txt:9]: `ConditionCompare`

File: forge-ai/src/main/java/forge/ai/ComputerUtilCombat.java [hunk 2]
```diff
@@ -2480,6 +2480,8 @@ public static int predictPoisonFromTriggers(Card attacker, Player attacked, int
                 }
                 // TODO: Predict replacement effects for counters (doubled, reduced, additional counters, etc.)
             }
+            // intern toxic effect
+            poison += attacker.getKeywordMagnitude(Keyword.TOXIC);
```
Comment (tool4ever) [inline forge-ai/src/main/java/forge/ai/ComputerUtilCombat.java:2484]: `hasDoubleStrike()` should probably be moved to the bottom so it only gets done once and also counts for Toxic

But I can do that later if you want

**Review by tool4ever — APPROVED** [review]

**Pull request merged**